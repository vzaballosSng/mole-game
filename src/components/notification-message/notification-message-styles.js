import { css } from 'lit';

export const notificationMessageStyles = css`
	:host {
    background-color: #050605a3;
    bottom: 0;
    color: white;

		height: 1.875rem;
    opacity: 0;
    padding: 1.25rem;
    position: fixed;
    width: 100%;
	}

  :host([show]) {
    opacity: 1;
    transition: opacity 1s ease;
  }

  .message {
    margin: 0;
    line-height: 2rem;
  }
`;
