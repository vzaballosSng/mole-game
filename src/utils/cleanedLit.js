// eslint-disable-next-line no-undef
const { prettyDOM } = require('@testing-library/dom');

export const cleanedLit = (htmlElement) => {
	const domHTML = prettyDOM(htmlElement, Infinity);
	const cleaned = domHTML.replace(/"<!--\?lit\$.*?-->"/g, '""');
	return cleaned;
};
export const cleanedLitElements = (htmlElements) => {
	const cleaned = htmlElements.map((htmlElement) => {
		return cleanedLit(htmlElement);
	});
	return cleaned;
};
