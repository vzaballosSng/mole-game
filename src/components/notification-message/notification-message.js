import { LitElement, html } from 'lit';
import { notificationMessageStyles } from './notification-message-styles.js';

export class NotificationMessage extends LitElement {
  static get properties() {
    return {
      message: { type: String },
      show: { type: Boolean }
    };
  }

  static get styles() {
    return [notificationMessageStyles];
  }

  constructor() {
    super();
    this.message = '';
  }

  render() {
    return html`<p class="message">${this.message}</p>
   `;
  }
}

window.customElements.define('notification-message', NotificationMessage);

