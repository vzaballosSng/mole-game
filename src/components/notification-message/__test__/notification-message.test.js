import { html, fixture } from '@open-wc/testing';
import { expect } from '@jest/globals';
import { axe, toHaveNoViolations } from 'jest-axe';
import { cleanedLit } from '../../../utils/cleanedLit.js';
import '../notification-message.js';

expect.extend(toHaveNoViolations);

describe('notification-message', () => {
	it('notification-message matches the snapshot', async () => {
		const el = await fixture(
			html`<notification-message></notification-message>`
		);
		const dataCleaned = cleanedLit(el);
		expect(dataCleaned).toMatchSnapshot();
	});

	it('accessibility test', async () => {
		const el = await fixture(html`<notification-message></notification-message>`);
		const results = await axe(el);
		expect(results).toHaveNoViolations();
	});
});
