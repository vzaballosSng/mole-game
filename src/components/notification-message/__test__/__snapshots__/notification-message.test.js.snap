// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`notification-message notification-message matches the snapshot 1`] = `"[36m<notification-message />[39m"`;
